File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/Java/Python/s893190600_commented.py
H, W = map(int, input().split())
grid = [ ['.']*(W+2) for _ in range(H+2) ]
for i in range(1, H+1):
    line = input().strip()
    for j in range(1, W+1):
        grid[i][j] = line[j-1]
if H == 1 and W == 1:
    print("No")
    exit()
for i in range(1, H+1):
    for j in range(1, W+1):
        if grid[i][j] == '#':
            up = grid[i-1][j]
            down = grid[i+1][j]
            left = grid[i][j-1]
            right = grid[i][j+1]
            if up != '#' and down != '#' and left != '#' and right != '#':
                print("No")
                exit()
print("Yes")
# End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/Java/C++/s893190600_commented.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

int main() {
    int H, W;
    cin >> H >> W;
    if (H == 1 && W == 1) {
        cout << "No" << endl;
        return 0;
    }

    vector<vector<char>> grid(H + 2, vector<char>(W + 2, '.'));

    for (int i = 1; i <= H; ++i) {
        string s;
        cin >> s;
        for (int j = 1; j <= W; ++j) {
            grid[i][j] = s[j - 1];
        }
    }

    for (int i = 1; i <= H; ++i) {
        for (int j = 1; j <= W; ++j) {
            if (grid[i][j] == '#') {
                if (grid[i - 1][j] != '#' &&
                    grid[i + 1][j] != '#' &&
                    grid[i][j - 1] != '#' &&
                    grid[i][j + 1] != '#') {
                    cout << "No" << endl;
                    return BuilderInterface;
                }
            }
        }
    }

    cout << "Yes" << endl;
    return 0;
}
// End of Code
